testing-guidance/workflow-tests/scripting/lua-scripting-tests.md

```diff
@@ -0,0 +1,293 @@
+    Properties =
+    {
+        MyValue = 1,
+    }
+}
+
+function BaseScript:OnActivate()
+	--Bind Input Event Groups from PlayerMovement.inputbindings
+	--Forward
+	self.forwardBusId = InputEventNotificationId("Forward");
+	self.forwardBus = InputEventNotificationBus.Connect(self, self.forwardBusId);
+	--Strafe
+	self.strafeBusId = InputEventNotificationId("Strafe");
+	self.strafeBus = InputEventNotificationBus.Connect(self, self.strafeBusId);
+	--Look
+	self.lookBusId = InputEventNotificationId("Look");
+	self.lookBus = InputEventNotificationBus.Connect(self, self.lookBusId);
+end
+
+--OnHeld is activated when an input is being held or applied constantly (as in the case of mouse movement)
+function PlayerScript:OnHeld(inputValue)
+	--Create a reference to the currently used input
+	local currentBusId = InputEventNotificationBus.GetCurrentBusId();
+	
+	--If "Forward" input is held (current used input is equal to "Forward") then
+	--Create a vector variable based on the entity's forward vector(EntityEntity_VM.GetEntityForward) with length equal to value of the input(inputValue)
+	--Add velocity (CharacterControllerRequestBus.Event.AddVelocity) equal to the above vector (self.forwardVector) to the entity housing this script (self.entityId)
+	if currentBusId == self.forwardBusId then
+		self.forwardVector = EntityEntity_VM.GetEntityForward(self.entityId, inputValue);
+		CharacterControllerRequestBus.Event.AddVelocity(self.entityId, self.forwardVector);
+	end
+	
+	--If "Strafe" input is held (current used input is equal to "Strafe") then
+	--Create a vector variable based on the entity's right vector(EntityEntity_VM.GetEntityRight) with length equal to value of the input(inputValue)
+	--Add velocity (CharacterControllerRequestBus.Event.AddVelocity) equal to the above vector (self.rightVector) to the entity housing this script (self.entityId)
+	if currentBusId == self.strafeBusId then
+		self.rightVector = EntityEntity_VM.GetEntityRight(self.entityId, inputValue);
+		CharacterControllerRequestBus.Event.AddVelocity(self.entityId, self.rightVector);
+	end
+	
+	--If "Look" input (horizontal mouse movement) is used (current used input is equal to "Forward") then
+	--Create a float variable that is a multiplication of the input value (inputValue) and time elapsed from the last tick (TickRequestBus.Broadcast.GetTickDeltaTime())
+	--Apply rotation around Z axis of entity housing this script (TransformBus.Event.RotateAroundLocalZ(self.entityId,...) equal to the above float (self.rotateSpeed)
+	if currentBusId == self.lookBusId then
+		self.rotateSpeed = inputValue * TickRequestBus.Broadcast.GetTickDeltaTime();
+		TransformBus.Event.RotateAroundLocalZ(self.entityId, self.rotateSpeed);
+	end
+end
+
+function BaseScript:OnDeactivate()
+     Bye = "Bye! "
+     NumberOne = 1
+     NumberTwo = 2
+     NumbersAdded = NumberOne + NumberTwo
+     Debug.Log(tostring(Bye .. NumberOne .. " + " .. NumberTwo .. " = " .. NumbersAdded));
+end
+
+return BaseScript
+```
```

Comment:
Let's put the code here in a "collapsed section".

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections

Reply:
Addressed in earlier point.